app/commands/contributor_team/membership/destroy.rb

```diff
@@ -15,7 +15,7 @@ def call
     private
     memoize
     def remains_in_team_role?
-      user.teams.any? { |t| t.role_for_members == team.role_for_members }
+      user.teams.map(&:role_for_members).include?(team.role_for_members)
```

Comment:
Using `map` here is the same as using `any` above in terms of it making all the Team objects. Using `pluck` just retrieves that one column from the database.

The only time this would be preferably if `user.teams` is already loaded elsewhere. Is that why you went for map?